executor/runtime/docker/docker.go

```diff
@@ -2279,7 +2282,19 @@ func (r *DockerRuntime) handleDockerEvent(message events.Message, statusMessageC
 		if exitCode == "0" {
 			statusMessageChan <- runtimeTypes.StatusMessage{
 				Status: runtimeTypes.StatusFinished,
-				Msg:    cName + " container successfully exited with 0",
+				Msg:    cName + " container successfully finished with exit code 0",
+			}
+			return isTerminalDockerEvent
+		} else if exitCode == "127" {
+			// An exit code of 127 means command not found.
+			// This isn't a very useful exit code for users, because it doesn't say exactly "What" the
+			// command was that was actually at play!
+			// We can help the user out a little bit by simply letting them know the command was.
+			firstCommand, fullCommand := r.GetFullCommandForDebug(cName)
+			msg := fmt.Sprintf("%s container couldn't find the command '%s' to run in the path (exit code %s) or a shell script returned the same thing. Full command: %s", cName, firstCommand, exitCode, fullCommand)
```

Comment:
For the nested error case, it looks like there'd be something like this in stderr. Could also mention something about checking stderr:
```
$ ./test.sh
./test.sh: line 3: missingcommand: command not found
```

Maybe something like this? I dunno...:
> [foo] container had a command not found error (exit code 127), either from the container command '[bar]' or from a shell script executed by that command. stderr may have more detail. Full command: [bar baz]

Reply:
If we wanted to go really overboard, we have the filesystem, we could start another container and inspect it. If the original entrypoint is there and executable, we can render the error about a shell script, if it's not present, then we can just say something about "this entrypoint doesn't exist in this image". IMO this as-is is fine for now.

Reply:
Yea, no need to go too crazy on this first pass.
However I would like to enhance the error messages on failed tasks with the last line of stdout/err, but in a different PR (more involved)